src/otx/algorithms/visual_prompting/tasks/inference.py

```diff
@@ -111,33 +119,29 @@ def get_config(self) -> Union[DictConfig, ListConfig]:
         self.hyper_parameters: VisualPromptingBaseConfig = self.task_environment.get_hyper_parameters()
 
         # set checkpoints
-        resume_from_checkpoint = model_checkpoint = None
-        if self.task_environment.model is not None:
-            # self.task_environment.model is set for two cases:
-            # 1. otx train : args.load_weights or args.resume_from
-            #   - path and resume are set into model_adapters
-            # 2. otx eval
-            #   - both resume_from and path are not needed to be set, path is not set into model_adapters,
-            #     so it can be distinguished by using it
-            resume_from_checkpoint = model_checkpoint = self.task_environment.model.model_adapters.get("path", None)
-            if isinstance(resume_from_checkpoint, str) and resume_from_checkpoint.endswith(".pth"):
-                # TODO (sungchul): support resume from checkpoint
-                logger.info("[*] Pytorch checkpoint cannot be used for resuming. It will be supported.")
-                resume_from_checkpoint = None
-            elif not self.task_environment.model.model_adapters.get("resume", False):
+        model_checkpoint: Optional[str] = None
+        resume_from_checkpoint: Optional[str] = None
+        if self.mode == "train" and self.task_environment.model is not None:
+            # when args.load_weights or args.resume_from is set
+            resume_from_checkpoint = model_checkpoint = self.task_environment.model.model_adapters.get("path", None)  # type: ignore  # noqa: E501
+            if self.task_environment.model.model_adapters.get("resume", False):
+                if resume_from_checkpoint.endswith(".pth"):  # type: ignore
```

Comment:
It's just a question. In other mmXX tasks, weight saved in the middle of training also can be used for resume. Isn't pytorch lightning model possible?

Reply:
It's possible. Lightning saves checkpoint in the middle of training and we can use these checkpoints to resume training.
A different thing from mmx is "just using path". Lightning trainer loads checkpoint and meta and enables to resume training.
There are too many considerations to manually load & apply checkpoint and meta.

Reply:
I understand. As you said, it'll be supported soon right?

Reply:
Resuming w/ lightning checkpoint is already available.

Reply:
Ah yes but it doesn't resume with `weights.pth` because file with "pth" suffix doesn't support resume. Then resume with "weights.pth" will be supported. Do I correctly understand it?

Reply:
Ah, sorry for my misunderstanding.
That will be supported, and metadata and suffix might be changed just for pytorch lightning.